### PR TITLE
Documentation for BigMod Algorithm

### LINKED ISSUE #474: Doc for Bigmod algorithm
## 📚 Documentation

(A clear and concise description of what the issue is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

YES

### PR DESCRIPTION
**Pull request by RaiyanMahin**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES
### Description

Documentation for BigMod Algorithm

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #474

### CODE REVIEW

**Comment by RaiyanMahin** [comment]
@iamrajiv @HarshCasper  @vybhav72954 kindly review

File: docs/competitive-programming/bigmod-algorithm.mdx [hunk 1]
```diff
@@ -0,0 +1,206 @@
+---
+title: Bigmod Algorithm
+slug: /competitive-programming/bigmod-algorithm
+---
+
+## Introduction
+
+In this article, we will learn about Bigmod Algorithm.
+Let us say, You want to calculate the value of `2^60`. This can be calculated by applying Bigmod algorithm.
+
```
Comment (iamrajiv) [inline docs/competitive-programming/bigmod-algorithm.mdx:10]: Remove this line break.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: docs/competitive-programming/bigmod-algorithm.mdx (hunk 1 above)
Comment (RaiyanMahin) [reply docs/competitive-programming/bigmod-algorithm.mdx]: done

**Comment by RaiyanMahin** [comment]
@atarax665 @iamrajiv  kindly review my PR

**Review by iamrajiv — APPROVED** [review]

**Comment by RaiyanMahin** [comment]
@atarax665 @ankitaggarwal23 @gat786 kindly review my PR

**Review by vybhav72954 — APPROVED** [review]

**Comment by RaiyanMahin** [comment]
@vybhav72954 if all things are good, please merge this PR

**Comment by RaiyanMahin** [comment]
@iamrajiv  add labels please

**Pull request merged**